test/recompilation-tests/hardhat-test/check.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+CONTRACT_NAME=contract_test_cache.cairo
+CONTRACT_PATH="contracts/${CONTRACT_NAME}"
+
+DEPENDENCY_NAME=dependency.cairo
+DEPENDENCY_PATH="contracts/${DEPENDENCY_NAME}"
+
+rm -rf $DEPENDENCY_PATH $CONTRACT_PATH
+# Hardhat test command
+echo "Testing Recompilation with new contract added"
+cp "$(dirname "$0")/$CONTRACT_NAME" "$CONTRACT_PATH"
+cp "$(dirname "$0")/$DEPENDENCY_NAME" "$DEPENDENCY_PATH"
+npx hardhat test --no-compile test/recompilation/recompilation-main-test.ts
+
+echo "Testing Recompilation with artifacts deleted"
+rm -rf starknet-artifacts/contracts/contract.cairo
+npx hardhat test --no-compile test/recompilation/recompilation-main-test.ts
+
+echo "Testing Recompilation with updated contract"
+# Appending a new function to the contract
+cat "$(dirname "$0")/get_balance.cairo" >> contracts/contract_test_cache.cairo
+npx hardhat test --no-compile test/recompilation/recompilation-update-test.ts
+
+echo "Testing Recompilation with cache file deleted"
+rm -rf cache/cairo-files-cache.json
+npx hardhat test --no-compile test/recompilation/recompilation-main-test.ts
+
+echo "Testing Recompilation with dependency changed"
+echo "#" >> "$DEPENDENCY_PATH"
+npx hardhat test --no-compile test/recompilation/recompilation-dependency-test.ts
+
+echo "Testing Recompilation with source deleted"
+CACHE_CONTENT_BEFORE=$(cat cache/cairo-files-cache.json)
+rm -rf contracts/contract_test_cache.cairo
+npx hardhat test --no-compile test/recompilation/recompilation-main-test.ts
+# Check that the cache file was updated
+CACHE_CONTENT_AFTER=$(cat cache/cairo-files-cache.json)
+if [ "$CACHE_CONTENT_BEFORE" == "$CACHE_CONTENT_AFTER" ]; then
```

Comment:
It would be better to test file equality with the `diff` command. It returns a non-zero status code if files are different and also displays the difference. You would probably only have to store the cairo-files-cache before running `hardhat test`.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/106/commits/aae8db5f05d4275b1146f5192fa128877a307479